server/controllers/project.controller/getProjectsForUser.js

```diff
@@ -10,25 +10,44 @@ import { toApi as toApiProjectObject } from '../../domain-objects/Project';
 const createCoreHandler = (mapProjectsToResponse) => async (req, res) => {
   try {
     const { username } = req.params;
+    const currentUser = req.user?._id || '';
+
     if (!username) {
       res.status(422).json({ message: 'Username not provided' });
       return;
     }
+
     const user = await User.findByUsername(username);
     if (!user) {
       res
         .status(404)
         .json({ message: 'User with that username does not exist.' });
       return;
     }
+
     const projects = await Project.find({ user: user._id })
       .sort('-createdAt')
-      .select('name files id createdAt updatedAt')
+      .select('name files id createdAt updatedAt visibility')
       .exec();
-    const response = mapProjectsToResponse(projects);
+
+    const publicProjectsOnly = projects.filter(
+      (project) => project.visibility === 'Public'
+    );
+
+    if (!req.user) {
+      res.json(publicProjectsOnly);
+      return;
+    }
+
+    const response = mapProjectsToResponse(
+      !currentUser.equals(user._id) ? publicProjectsOnly : projects
+    );
+
     res.json(response);
-  } catch (e) {
-    res.status(500).json({ message: 'Error fetching projects' });
+  } catch (error) {
+    res
+      .status(500)
+      .json({ success: false, message: 'Error fetching projects' });
```

Comment:
> I'm going to see if I can figure out why you're getting errors in the `getProjectsForUser` handler. 

The test failure is something really simple.  You added `{ success: false }` to the 500 error response so the `function expectations()` in the test file needs to be updated to include `{ success: false }`.  We are not very consistent with the formatting of error responses and that is something that I would like to address at some point.

Reply:
Oh, wowwww! Although I know a good practice to set up default formatting for error responses (as well as success responses) rather than changing everywhere independently, we can create a custom class for both types of responses and use them all over the backend, overwriting values accordingly for errors. this way it will be a consistent responses?!

```
class ApiError  {
    constructor(
        statusCode,
        message= "Something went wrong",
        errors = [],
        stack = ""
    ){
        super(message)
        this.statusCode = statusCode
        this.data = null
        this.message = message
        this.success = false;
        this.errors = errors
    }
}
```